console/v1/types_console_quick_start.go

```diff
@@ -0,0 +1,93 @@
+	// tags is an array of strings that describe the Quick Start.
+	// +optional
+	Tags []string `json:"tags,omitempty"`
+	// duration informs user on aproximately how long the Quick Start should take to finish.
+	// +optional
+	Duration int `json:"duration,omitempty"`
+	// description is the description of the Quick Start. (includes markdown)
+	// +optional
+	Description string `json:"description,omitempty"`
+	// prerequisites contains a list prerequisites for the Quick.
+	// +optional
+	Prerequisites []string `json:"prerequisites,omitempty"`
+	// introduction describes the purpose and usability of the Quick Start. (includes markdown)
+	Introduction string `json:"introduction"`
+	// tasks contains list of tasks that user has to take in order to pass the Quick Start.
+	Tasks []ConsoleQuickStartTask `json:"tasks"`
+	// conclusion sums up the Quick Start and suggests the possible
+	// next Quick Starts for user to take. (includes markdown)
+	// +optional
+	Conclusion string `json:"conclusion,omitempty"`
+	// nextQuickStart is the name of the following Quick Start.
+	// +optional
+	NextQuickStart string `json:"nextQuickStart,omitempty"`
+	// accessReviewResources containes arrays of resources that user's access
+	// will be reviewed against in order for the user to take the Quick Start.
+	// +optional
+	AccessReviewResources []authorizationv1.ResourceAttributes `json:"accessReviewResources,omitempty"`
+}
+
+// ConsoleQuickStartTask contains Quick Start's task, which instructs user to
+// through the Console UI in order to operate with necessary resource.
+type ConsoleQuickStartTask struct {
+	// title describes the task and will be rendered as a step header.
+	Title string `json:"title"`
+	// description contains task's description, necessary to pass the task. (includes markdown)
+	Description string `json:"description"`
+	// review contains instructions that validate taken. Based on the outcome of the
+	// review steps user will have to select 'Yes' or 'No' radio button, which indicates
+	// whether the step was passed succesfully or unsuccesfully.
+	// +optional
+	Review *ConsoleQuickStartTaskReview `json:"review,omitempty"`
+	// recapitulation contains information about passed step.
+	// +optional
+	Recapitulation *ConsoleQuickStartTaskRecapitulation `json:"recapitulation,omitempty"`
+}
+
+// ConsoleQuickStartTaskReview contains instructions that validate taken. Based on the
+// outcome of the review steps user will have to select 'Yes' or 'No' radio button,
+// which indicates whether the step was passed succesfully or unsuccesfully.
+type ConsoleQuickStartTaskReview struct {
+	// instructions contains steps that user needs to take in order
+	// to validate his work after going through a task. (includes markdown)
+	Instructions string `json:"instructions"`
+	// taskHelp contains suggestion with how to deal with failed task
+	// review and will will be rendered at the end of task.
+	TaskHelp string `json:"taskHelp"`
```

Comment:
We might want to call this `failedTaskHelp` to make it clear it will only be displayed if not successful.